controllers/logout_test.go

```diff
@@ -0,0 +1,43 @@
+package controllers
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/gutakk/go-google-scraper/tests"
+	"github.com/stretchr/testify/suite"
+	"gopkg.in/go-playground/assert.v1"
+)
+
+type LogoutTestSuite struct {
+	suite.Suite
+	engine *gin.Engine
+}
+
+func (s *LogoutTestSuite) SetupTest() {
+	s.engine = tests.GetRouter(false)
+	new(LogoutController).applyRoutes(EnsureAuthenticatedUserGroup(s.engine))
+}
+
+func TestLogoutTestSuit(t *testing.T) {
+	suite.Run(t, new(LogoutTestSuite))
+}
+
+func (s *LogoutTestSuite) TestLogoutWithAuthenticatedUser() {
+	cookie := "go-google-scraper=MTYwNjQ2Mjk3MXxEdi1CQkFFQ180SUFBUkFCRUFBQUlmLUNBQUVHYzNSeWFXNW5EQWtBQjNWelpYSmZhV1FFZFdsdWRBWUVBUDRFdFE9PXzl6APqAQw3gAQqlHoXMYrPpnqPFkEP8SRHJZEpl-_LDQ=="
+	headers := http.Header{}
+	headers.Set("Cookie", cookie)
+
+	response := tests.PerformRequest(s.engine, "POST", "/logout", headers, nil)
+
+	assert.Equal(s.T(), http.StatusFound, response.Code)
+	assert.Equal(s.T(), "/", response.Header().Get("Location"))
+}
+
+func (s *LogoutTestSuite) TestLogoutWithGuestUser() {
+	response := tests.PerformRequest(s.engine, "POST", "/logout", nil, nil)
+
+	assert.Equal(s.T(), http.StatusFound, response.Code)
+	assert.Equal(s.T(), "/login", response.Header().Get("Location"))
```

Comment:
I don't see how this test tell us anything as it has the same assertions as `TestLogoutWithAuthenticatedUser` 🙃  Since logout requires an authenticated user, it could throw a 401 error if it was an API but for a web application, setting a flash and redirect is better. So how about adding an assertion for the flash message?

Reply:
As per we discussed in slack, added the todo comment in fac19d0124358e3173582b65d7a05b5a64f2c538